webpack/ForemanTasks/Components/TasksTable/TasksTableActions.js

```diff
@@ -94,29 +95,76 @@ export const unselectRow = id => ({
   payload: id,
 });
 
-export const actionSelected = (actionType, selected, url) => async dispatch => {
+export const bulkResumeRequest = resumeTasks => {
+  const ids = resumeTasks.map(task => task.id);
+  const url = new URI('/foreman_tasks/api/tasks/bulk_resume');
+  url.setSearch('task_ids[]', ids);
+  return API.post(url);
+};
+
+export const bulkResume = (selected, currentURL) => async dispatch => {
+  const resumeTasks = selected.filter(task => task.isResumeble);
+  if (resumeTasks.length < selected.length)
+    dispatch(
+      addToast({
+        type: 'warning',
+        message: __('Not all the selected tasks can be resumed'),
+      })
+    );
+  if (resumeTasks.length) {
+    try {
+      const { data } = await bulkResumeRequest(resumeTasks);
```

Comment:
In my Redux dev tools, I see `TOASTS_ADD` actions but no API request actions.  The actual API calls are taking place in these side effect functions (`bulkResumeRequest` and `bulkCancelRequest`).

I suggest dispatching `TASKS_REQUEST` and `TASKS_SUCCESS`/`TASKS_FAILURE` actions to document that API calls are actually taking place.  (Whether you differentiate between bulk and regular requests I'll leave up to you)

Reply:
I can do this but can you please explain why do we want to document API calls in Redux?

Reply:
Having actions that document what happened, even if they don't change the state, is still quite useful in Redux.   These "document actions" help provide one of the major benefits of Redux, which is that you can look at Redux Devtools and know everything that happened in your app, and in what order.  This can be very useful for troubleshooting (did this API request happen before I was done gathering my data?  Did I send duplicate requests by mistake? Is the API request happening at all?)  With these "document actions" I can know the answers to all of these just by looking at the Redux devtools.

In Foreman our new API middleware watches for the document action (`API_OPERATIONS.GET` action type) and makes the actual API call in the middleware.  I think this is a great pattern; however this currently only handles GET requests and not POST.

Reply:
Added thanks for the explanation